spec/src/main/asciidoc/core/spi.asciidoc

```diff
@@ -1580,3 +1594,35 @@ CDI 2.0 introduced the following Configurators interface:
 * <<observer_method_configurator>> for `ObserverMethod` configuration
 
 The container must provide implementation for all these configurators and make them available in matching container lifecycle events as defined in <<init_events>>.
+
+[[interception_proxy_factory]]
+
+=== Apply interceptor and decorators programmatically
+
+Contextual instances of bean having interceptors binding or defined decorators are provided by the container as a proxy.
```

Comment:
On a second thought if we go in that direction (no mention of proxy) we should perhaps find another name for `InterceptionProxyFactory` interface, something like `ContextualReferenceFactory` or `InterceptedInstanceFactory`?
wdyt?